uaclient/testing/fakes.py

```diff
@@ -34,3 +36,47 @@ def request_url(
         if isinstance(response, Exception):
             raise response
         return response, {"header1": ""}
+
+
+class MockEntitlement(base.UAEntitlement):
```

Comment:
I think the idea here is a good one, like overriding the entitlement status methods, like `application and applicability`, but we already have a similar entitlement factory as can be seen here:

https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/entitlements/tests/conftest.py#L83

I think we bring some of those changes there

Reply:
The reason I couldn't use that factory is that it creates an entitlement _object_, but for this test, we need the _class_.
The factory also requires an actual entitlement class to instantiate, but this test isn't testing any particular entitlement, so we'd have to just pick one of esm-apps, livepatch, etc. - which is fine, just perhaps misleading to instantiate a specific subclass when the specific subclass doesn't matter.

I based `MockEntitlement` off of the `ConcreteTestEntitlement` found in `entitlements/tests/test_base.py`. I didn't reuse `ConcreteTestEntitlement` because the tests in `test_base.py` relied on `ConcreteTestEntitlement` actually looking at the `machine-token.json` for entitlement status, which seemed like too much complexity for this test.

Ideally this test would get a class that inherits from `UAEntitlement` and implements the bare minimum so that it can be instantiated. Ideally it would have no side affects or dependencies on files, and it wouldn't actually do anything when "enabled" or "disabled". That's the role this MockEntitlement is supposed to fill.

I can try to use the entitlement factory that you point too though if you'd prefer

Reply:
@lucasmoura This particular test I'm adding in `test_cli_enable.py` is just weird because, unlike all the other tests in that file, it relies on specific functionality in `UAEntitlement`, which complicates everything. I'm tempted to just remove this test since we have the integration test for this scenario. Then i could get rid of this MockEntitlement, but we would keep the `enable`/`_perform_enable` refactor - thoughts?

Reply:
My mistake again, I forgot that we cannot directly instantiate the `UAEntitlement`  base class, so I agree the `entitlement_factory` is not the right place for it.

Regarding the `MockEntitlement`  abstraction, I think we could use the `ConcreteTestEntitlement` code here, like move it to the `fakes.py` module but we would not need to move the `factory` method there as well. I mean, the `MockEntitlement` would be the `ConcreteTestEntitlement` and the `test_base` would just use it in the factory method.

But, I do agree that this is turning into a huge overkill over a test that will not bring us any major contribution, since, as you said, we do have an integration test for it. So, if you want to remove it, I don't see a problem here. Also, sorry for the overkill here, I honestly thought that this test was going to be simpler to implement.

Reply:
@lucasmoura Thanks for all your feedback through this. I ended up drastically reducing the scope of the test, so it is just testing to make sure "One moment, checking your subscription" is printed and not printing any other result, which is handled by `can_enable` for this case. That removed the need for the MockEntitlement all together.

So if Jenkins is good, I think I'm happy with this PR (but if integration tests don't pass then I'll take a look of course).